packages/frontend/src/components/RichTextEditor/MenuBar.tsx

```diff
@@ -190,35 +224,147 @@ interface MenuBarProps {
+    [MenuLabels.ImageAdd]:
+      'Enter direct image link (e.g. https://file.go.gov.sg/clipplumer.png)',
+  }
   if (!editor) {
     return null
   }
 
   return (
-    <div className="editor__header">
-      {menuButtons.map(({ onClick, label, icon, isActive }, index) => {
-        if (!onClick) {
-          return <div className="divider" key={`${label}${index}`} />
-        }
-        return (
-          <button
-            key={`${label}${index}`}
-            title={label}
-            style={{
-              borderRadius: '0.25rem',
-              width: 'auto',
-              minWidth: 0,
-              backgroundColor: isActive?.(editor)
-                ? 'rgba(0,0,0,0.1)'
-                : 'transparent',
-            }}
-            className={`menu-item${isActive?.(editor) ? ' is-active' : ''}`}
-            onClick={() => onClick(editor)}
+    <>
+      <div className="editor__header">
+        {menuButtons.map(({ onClick, label, icon, isActive }, index) => {
+          if (!onClick) {
+            return <div className="divider" key={`${label}${index}`} />
+          }
+          return (
+            <button
+              key={`${label}${index}`}
+              title={label}
+              style={{
+                borderRadius: '0.25rem',
+                width: 'auto',
+                minWidth: 0,
+                backgroundColor: isActive?.(editor)
+                  ? 'rgba(0,0,0,0.1)'
+                  : 'transparent',
+              }}
+              className={`menu-item${isActive?.(editor) ? ' is-active' : ''}`}
+              onClick={() => {
+                if (onClickOverrides && onClickOverrides[label]) {
+                  onClickOverrides[label]()
+                  return
+                }
+                onClick(editor)
+              }}
+            >
+              {icon}
+            </button>
+          )
+        })}
+      </div>
+      <Dialog
+        open={showValueDialog}
+        onClose={() => setShowValueDialog(false)}
+        PaperProps={{
+          // so that the variable selector float can overlay
+          sx: { display: 'inline-table' },
+        }}
+      >
+        <DialogTitle>{dialogLabel}</DialogTitle>
+        <DialogContent>
+          <DialogContentText
+            tabIndex={-1}
+            component="div"
+            className="menubar-dialog-content"
           >
-            {icon}
-          </button>
-        )
-      })}
-    </div>
+            {dialogLabel && (
+              <Form onSubmit={() => dialogOnSubmits[dialogLabel]()}>
+                <BareEditor
+                  // val is in HTML, need to parse back to plain text
+                  onChange={(val) => setDialogValue(parse(val).textContent)}
+                  initialValue={dialogValue}
+                  editable
+                  variablesEnabled
+                  placeholder={dialogPlaceholders[dialogLabel]}
+                />
+                <LoadingButton
```

Comment:
is it possible to disable if the content is empty? ignore if its too difficult or else can just tell user in the `dialogPlaceholder` that empty link will unset the url